src/borg/repository.py

```diff
@@ -190,6 +190,9 @@ def load_key(self):
         return keydata.encode('utf-8')  # remote repo: msgpack issue #99, returning bytes
 
     def get_free_nonce(self):
+        if not self.lock.got_exclusive_lock():
+            raise AssertionError("bug in code, exclusive lock should exist here")
```

Comment:
I copied from prepare_txn, but yes, here the situation is simpler so i’ll just use a normal assert. Do we worry about python removing these checks in optimize mode? That would be the only reason to keep it as explicit raise.